Extension/src/LanguageServer/utils.ts

```diff
@@ -112,3 +112,9 @@ export async function withCancellation<T>(promise: Promise<T>, token: vscode.Can
         });
     });
 }
+
+export async function checkTime<T>(fn: () => Promise<T>): Promise<{ result: T; time: number }> {
+    const start = Date.now();

```

Comment:
(non-blocking) If you're expecting to use this checkTime function in very fast or performance-critical contexts, consider using [`performance.now()`](https://nodejs.org/api/perf_hooks.html#performancenow) instead of `Date.now()`. It provides sub-millisecond precision.

Reply:
It currently outputs number of milliseconds. I'm doing think sub-millisecond precision is important in this case.

Reply:
Changed to [performance.now()](https://nodejs.org/api/perf_hooks.html#performancenow), clients can decide the precision they want to use.